src/app/api/v1/premium/route.ts

```diff
@@ -10,20 +10,13 @@ export async function GET(req: NextRequest) {
     // NOTE: the email address passed here cannot
     // be trusted, we must also check that the
     // `subscriptions` claim on the FxA token contains
-    // "monitor".
+    // "monitor", or call the auth server API to confirm.
```

Comment:
One thought, if we do need to keep using this API long-term: we might want to do the API call here, and then record in the database that this user is a subscriber. We'd still need to periodically check that the user is still subscribed, and take action, but we could record a timestamp so we don't have to call it more often than once a day or so.

We might want to have a cron job that checks for unsubscribed users and deletes their data. Maybe we could email them first to let them know, and give a link to come back and renew if they want?